src/OrleansRuntime/Core/Dispatcher.cs

```diff
@@ -637,8 +655,10 @@ internal void SendMessage(Message message, ActivationData sendingActivation = nu
             // second, we check for a strategy associated with the target's interface. third, we check for a strategy associated with the activation sending the
             // message.
             var strategy = targetAddress.Grain.IsGrain ? catalog.GetGrainPlacementStrategy(targetAddress.Grain) : null;
+            var request = message.GetDeserializedBody(this.serializationManager) as InvokeMethodRequest;
```

Comment:
Likely more wasteful than the other instance, since there is little chance the new activation will be placed on the current silo (vs activation about to process the request)